pkg/systemd/services/service-details.jsx

```diff
@@ -208,20 +210,63 @@ export class ServiceDetails extends React.Component {
             waitsFileAction: false,
             note: "",
             error: "",
+            unit_properties: {},
         };
 
         this.onOnOffSwitch = this.onOnOffSwitch.bind(this);
         this.unitAction = this.unitAction.bind(this);
         this.unitFileAction = this.unitFileAction.bind(this);
+
+        this.unitType = props.unit.Id.split('.').slice(-1)[0];
+        this.unitTypeCapitalized = this.unitType.charAt(0).toUpperCase() + this.unitType.slice(1);
+        this.doMemoryCurrentPolling = this.doMemoryCurrentPolling.bind(this);
+
+        // MemoryCurrent property does not emit a changed signal - do polling for this property
+        if (props.unit.ActiveState == "active") {
+            this.doMemoryCurrentPolling();
+            this.interval = setInterval(this.doMemoryCurrentPolling, METRICS_POLL_DELAY);
+        }
+    }
+
+    componentDidUpdate(prevProps) {
+        // If unit became active start property polling and if got inactive stop
+        if (this.props.unit.ActiveState === 'active' && !this.interval) {
+            this.doMemoryCurrentPolling();
+            this.interval = setInterval(this.doMemoryCurrentPolling, METRICS_POLL_DELAY);
+        }
+        if (this.props.unit.ActiveState === 'inactive' && this.interval) {
+            this.doMemoryCurrentPolling();
+            clearInterval(this.interval);
+        }
+    }
+
+    componentWillUnmount() {
+        if (this.interval)
+            clearInterval(this.interval);
+    }
+
+    static getDerivedStateFromProps(nextProps, prevState) {
+        return {
+            waitsAction: nextProps.loadingUnits,
+            waitsFileAction: nextProps.loadingUnits,
+        };
+    }
+
+    doMemoryCurrentPolling() {
+        systemd_client.call(this.props.unit.path,
+                            "org.freedesktop.DBus.Properties", "Get",
+                            ["org.freedesktop.systemd1." + this.unitTypeCapitalized, 'MemoryCurrent'])
+                .then(result => {
+                    this.addUnitProperties(
+                        "MemoryCurrent",
+                        result[0] && result[0].v > 0 ? result[0].v : null,
+                    );
+                }, ex => console.log(ex.message));
```

Comment:
FWIW, the previous "Failed to.." string was fine, I meant to add `ex.toString()` as a second argument. But good enough.